makefiles/docker.inc.mk

```diff
@@ -92,18 +174,7 @@ DOCKER ?= docker
 ..in-docker-container:
 	@$(COLOR_ECHO) '$(COLOR_GREEN)Launching build container using image "$(DOCKER_IMAGE)".$(COLOR_RESET)'
 	$(DOCKER) run $(DOCKER_FLAGS) -t -u "$$(id -u)" \
-	    -v '$(RIOTBASE):$(DOCKER_BUILD_ROOT)/riotbase' \
-	    -v '$(RIOTCPU):$(DOCKER_BUILD_ROOT)/riotcpu' \
-	    -v '$(RIOTBOARD):$(DOCKER_BUILD_ROOT)/riotboard' \
-	    -v '$(RIOTMAKE):$(DOCKER_BUILD_ROOT)/riotmake' \
-	    -v '$(RIOTPROJECT):$(DOCKER_BUILD_ROOT)/riotproject' \
-	    -v /etc/localtime:/etc/localtime:ro \
-	    -e 'RIOTBASE=$(DOCKER_BUILD_ROOT)/riotbase' \
-	    -e 'CCACHE_BASEDIR=$(DOCKER_BUILD_ROOT)/riotbase' \
-	    -e 'RIOTCPU=$(DOCKER_BUILD_ROOT)/riotcpu' \
-	    -e 'RIOTBOARD=$(DOCKER_BUILD_ROOT)/riotboard' \
-	    -e 'RIOTMAKE=$(DOCKER_BUILD_ROOT)/riotmake' \
-	    -e 'RIOTPROJECT=$(DOCKER_BUILD_ROOT)/riotproject' \
+	    $(DOCKER_VOLUMES_AND_ENV) \
 	    $(DOCKER_ENVIRONMENT_CMDLINE) \
-	    -w '$(DOCKER_BUILD_ROOT)/riotproject/$(BUILDRELPATH)' \
+	    -w '$(DOCKER_APPDIR)' \
 	    '$(DOCKER_IMAGE)' make $(DOCKER_MAKECMDGOALS) $(DOCKER_OVERRIDE_CMDLINE)
```

Comment:
An idea to the future: Propagate the job count from the top level make to the docker make: `make -j9` etc.

Reply:
I tried in the past to get the job count from the main makefile but could not :/
I think we only have access to it from a sub make instance.

Reply:
Maybe we have it from within the target execution environment though as it is passed to sub make instances.

Reply:
Indeed  `MAKEFLAGS` has it when in the execution environment, and we could extract with a `$(filter -j%,$(MAKEFLAGS))` or alike.

Reply:
Some local tested solution for this, configured from outside so re-using `DOCKER_MAKECMDGOALS`.

```
# Extract parallel option
# The number of jobs is only available after make 4.2
# The value will not be available in make but in the targets body
MAKE_PARALLEL = $(filter -j -j%,$(MAKEFLAGS))
DOCKER_MAKECMDGOALS += $(MAKE_PARALLEL)
```

We would get `-j` even when saying `-j2` for make before `4.2` so would only pass `-j` in ubuntu bionic. But it could still be good.